### PR TITLE
Tag ModiaMath.jl v0.2.1

### PR DESCRIPTION
**Pull request by attobot**
Repository: [ModiaSim/ModiaMath.jl](https://github.com/ModiaSim/ModiaMath.jl)
Release: [v0.2.1](https://github.com/ModiaSim/ModiaMath.jl/releases/tag/v0.2.1)
Diff: [vs v0.2.0](https://github.com/ModiaSim/ModiaMath.jl/compare/e98462c9c57bbce3dbba3953611c5f1c003974fe...7bfc3c98b587d8bc34b01b801c9b0e8713de1ed5)
`requires` vs v0.2.0: no changes
cc: @MartinOtter

Please make sure that:
- CI passes for supported Julia versions (if applicable).
- Version bounds reflect minimum requirements.

### CODE REVIEW

**Comment by MartinOtter** [comment]
Hm. I do not know what to do. One test in JuliaCIBot (for Julia 0.7.0) fails, because in a test module "import ModiaMath" is made and this statement fails (but ModiaMath is the package that is tested). The second test in JuliaCIBot (for Julia 0.7.0) fails, because StaticArrays could not be loaded.

**Comment by andreasnoack** [comment]
Please add a Travis badge to your README such that it is easier to find the tests. It looks like your tests are failing there as well https://travis-ci.org/ModiaSim/Modia.jl. Please also enable testing on 1.0.

**Comment by andreasnoack** [comment]
Is this still relevant?

**Comment by MartinOtter** [comment]
Current status:
- Travis badge was added.
- Additional fix for Julia 1.0 (warn(..) -> @warn ...).
- The issue with the failed Travis CL was finally understood (package StaticArrays missing) and fixed in ModiaMath#master (see [discussion on discourse](https://discourse.julialang.org/t/travis-ci-fails-for-julia-v0-7-since-test-package-missing/13290/10?u=martinotter)).
- You wrote: *Please make sure that: Version bounds reflect minimum requirements*. It is not clear to me what to do here. My current assumption was that the REQUIRE file is used by Julia 0.6 (and in the REQUIRE file it is stated `julia 0.6`, so it should run with Julia 0.6 and later Julia versions) and that the Project.toml and Manifest.toml files are used by Julia 0.7 and 1.0. On the other hand [here](https://github.com/ModiaSim/ModiaMath.jl/pull/10) it is stated, that *Package registration currently still uses the old REQUIRE/METADATA system, and autogenerates the appropriate files for the new system. AFAIK, it'll be easier if package developers wait for the "official" migration to the new Project.toml/Manifest.toml format rather than trying to preemptively migrate on their own*. So, it seems that the Project.toml/Manifest.toml files should be removed?

To summarize, since there are changes to ModiaMath to fix the detected issues, I will make a new ModiaMath version 0.2.2 in a few days (documentation needs to be slightly adapted).

**This means that this Pull Request can be closed.**

**Comment by andreasnoack** [comment]
Thanks for the reply.

> So, it seems that the Project.toml/Manifest.toml files should be removed?

You can keep them but the REQUIRE files detemines what goes into the registry.

> To summarize, since there are changes to ModiaMath to fix the detected issues, I will make a new ModiaMath version 0.2.2 in a few days (documentation needs to be slightly adapted).

Instead of making a 0.2.2 release, I'd recommend just updating this release. See https://github.com/attobot/attobot#updatingdeleting-a-release

**Comment by MartinOtter** [comment]
> Instead of making a 0.2.2 release, I'd recommend just updating this release. See https://github.com/attobot/attobot#updatingdeleting-a-release

O.k. Will follow your advice.

**Comment by MartinOtter** [comment]
I looked at the failures of JuliaCIBot (note, the [Travis CL of ModiaMath](https://travis-ci.org/ModiaSim/ModiaMath.jl) is successful). For all Juila versions (0.6, 0.7, 1.0) the issue is that package PyPlot could not be properly installed. I do not know how I should fix this.

In the [Travis CL script](https://github.com/ModiaSim/ModiaMath.jl/blob/master/.travis.yml) this is performed by the line:

```
  - julia -e 'ENV["PYTHON"]=""; using Pkg; Pkg.add("Conda"); using Conda; Conda.update(); Conda.add("matplotlib"); Conda.add("basemap"); Pkg.add("PyCall"); Pkg.build("PyCall"); Pkg.add("PyPlot");'  
```

but it seems there is no possibility to add this to the JuliaCIBot?

One possibility would be that I rewrite the tests so that PyPlot is not called (this is not much work, because PyPlot is called via a special plot function and I could adapt this function) and remove PyPlot from the REQUIRE and Project.toml/Manifest.toml files. In that case the JuliaCIBot should go through. However, most likely, this will then not work for the end-user when he/she wants to get a plot


@ChrisRackauckas, @tshort, @crlaugh: Could you give an advice how I can fix this?

**Comment by tshort** [comment]
Longer term, I'd prefer removing the dependency on PyPlot. Instead, it'd be better to write recipes for Plots and/or Makie. That could be accomplished by integrating with JuliaDiffEq (which has Plots recipes).

In the short term, what's the harm in ignoring JuliaCIBot (and rely on Travis)?

**Comment by MartinOtter** [comment]
> In the short term, what's the harm in ignoring JuliaCIBot (and rely on Travis)?

A package seems to be only accepted in METADATA if the JuliaCIBot is successful.

**Comment by tshort** [comment]
A couple of projects have deleted `Project.toml` for CI Bot compatibility (e.g. https://github.com/JuliaAttic/Nullables.jl/pull/18). That's probably not the issue with PyPlot, though.

**Comment by andreasnoack** [comment]
> A package seems to be only accepted in METADATA if the JuliaCIBot is successful.

It's not as strict but if the CIBot tests point to an obvious problem then we usually ask that they are fixed. Sometimes there are tests that CIBot can't handle and then we rely on the Travis tests. In this case, we are hitting an issue where PyCall has problems finding `libpython` when using Python 3 which CIBot is. This shouldn't hold back this release. However, you specify that the minimum requirement for Julia is 0.6 but you only test version 1.0 on Travis so please either increase the Julia requirement or enable tests on all supported versions of Julia.

@tshort CIBot has now been upgraded (since the weekend) so it is now using the new package manager and, therefore, able to handle project files.

**Comment by MartinOtter** [comment]
> However, you specify that the minimum requirement for Julia is 0.6 but you only test version 1.0 on Travis so please either increase the Julia requirement or enable tests on all supported versions of Julia.

I have adapted Travis to test for Juila 0.6, 0.7, 1.0 (Linux and OSX). The tests are successful (see [log of travis file](https://travis-ci.org/ModiaSim/ModiaMath.jl))
